packages/subgraph/src/utils.ts

```diff
@@ -242,22 +242,20 @@ export function getAmountStreamedSinceLastUpdatedAt(
 }
 
 /**
- * calculateMaybeCriticalAtTimestamp will return optimistic date based on updatedAtTimestamp, totalDeposit, balanceUntilUpdatedAt and totalNetFlowRate.
+ * calculateMaybeCriticalAtTimestamp will return optimistic date based on updatedAtTimestamp, balanceUntilUpdatedAt and totalNetFlowRate.
  * @param updatedAtTimestamp
  * @param balanceUntilUpdatedAt
- * @param totalDeposit
  * @param totalNetFlowRate
  */
 
 export function calculateMaybeCriticalAtTimestamp(
     updatedAtTimestamp: BigInt,
     balanceUntilUpdatedAt: BigInt,
-    totalDeposit: BigInt,
     totalNetFlowRate: BigInt
 ): BigInt {
     if (balanceUntilUpdatedAt.le(BIG_INT_ZERO)) return BIG_INT_ZERO;
```

Comment:
Not sure I like this because when an account becomes critical then many streams can be liquidated which means that after the first liquidation this currently becomes zero when it's still critical and there are more streams to be liquidated.

I'd rather return the `updatedAtTimestamp` and only nullify this when the flow rate is zero or positive.

EDIT: Actually, I might use "0" to point to entities _already_ critical and use NULL to point to entities without a critical timestamp.

Thoughts? @0xdavinchee @msoni89

Reply:
So ideally the timestamp would convey 3 states:
1. Already critical
2. Will be critical
3. Not possible to be critical

Right now the solution does not convey 1. very well.

Reply:
@kasparkallas

I think it makes sense to do this:
1. Already critical, criticalAtTimestamp = 0 or `updatedAtTimestamp` (but this may not illustrate that the account was critical at a prior time)
2. Will be critical, criticalAtTimestamp = calculatedCriticalTimestamp
3. Not possible to be critical = null

I think 1. should be 0 or `updatedAtTimestamp`, maybe it's better if it's 0 because we can't guarantee that the liquidations are always happening on the same block that the account is critical and so we may have some incorrect information there

Reply:
Another option would be to keep the initial `maybeCriticalAtTimestamp` for 1.

Reply:
handled 3rd returning null when `Not possible to be critical`